alpaca/broker/client.py

```diff
@@ -216,7 +259,87 @@ def upload_cip_data_for_account_by_id(self, account_id: Union[UUID, str]):
+    def get_account_activities(
+        self,
+        activity_filter: GetAccountActivitiesRequest,
+        handle_pagination: Optional[PaginationType] = None,
+    ) -> Union[List[BaseActivity], Iterator[List[BaseActivity]]]:
+        """
+        Gets a list of Account activities, with various filtering options. Please see the documentation for
+        GetAccountActivitiesRequest for more information as to what filters are available.
+
+        The return type of this function is List[BaseActivity] however the list will contain concrete instances of one
+        of the child classes of BaseActivity, either TradeActivity or NonTradeActivity. It can be a mixed list depending
+        on what filtering criteria you pass through `activity_filter`
+
+
+        Args:
+            activity_filter (GetAccountActivitiesRequest): The various filtering fields you can specify to restrict
+              results
+            handle_pagination (Optional[PaginationType]): What kind of pagination you want. If None then defaults to
+              `PaginationType.FULL`
+
+        Returns:
+            Union[List[BaseActivity], Iterator[List[BaseActivity]]]: Either a list or an Iterator of lists of
+              BaseActivity child classes
+        """
+
+        if handle_pagination is None:
+            handle_pagination = PaginationType.FULL
+
+        # user wants no pagination, so just do a single request, convert the results we got and move on
+        if handle_pagination == PaginationType.NONE:
+            result = self.get(
+                "/accounts/activities", activity_filter.to_request_fields()
+            )
+            return [self._parse_activity(activity) for activity in result]
+
+        def get_as_iterator(
+            mapping: Callable[[HTTPResult], List[BaseActivity]]
+        ) -> Iterator[List[BaseActivity]]:
+            while True:
+                result = self.get(
+                    "/accounts/activities", activity_filter.to_request_fields()
+                )
+
+                # the api returns [] when it's done
+
+                if not isinstance(result, List) or len(result) == 0:
+                    break
+
+                yield mapping(result)
+
+        # otherwise, user wants pagination so we grab an interator
+        iterator = get_as_iterator(lambda x: [self._parse_activity(x) for x in x])
+
+        if handle_pagination == PaginationType.FULL:
+            # the iterator returns "pages", so we use chain to flatten them all into 1 list
+            return list(chain.from_iterable(iterator))
+        elif handle_pagination == PaginationType.ITERATOR:
+            return iterator
+
+    # ############################## HELPER/PRIVATE FUNCS ################################# #
+
+    @staticmethod
+    def _parse_activity(data: dict) -> Union[TradeActivity, NonTradeActivity]:
+        """
+        We cannae just use parse_obj_as for Activity types since we need to know what child instance to cast it into.
```

Comment:
`canne` looks like a typo

Reply:
Sorry I guess cannae might not make sense to most people :see_no_evil:. 

It's not a typo just scots for cannot. I'll fix it up